My recent changes and modularized main file.

jiljames opened the pull request:
Hey! Here is my changes for today's meeting. @mhrmm

Commit:
Trying out a change to the loss calculation.

jiljames committed:
Merge pull request #2 from jiljames/working

Trying out a change to the loss calculation.

Commit:
Changed haiku_helper to account for validation file

jiljames committed:
Merge pull request #3 from jiljames/working

Changed haiku_helper to account for validation file

Commit:
chenged vocab length in sequence_gan.py

jiljames committed:
Merge pull request #4 from jiljames/working

changed vocab length in sequence_gan.py

Commit:
Removed unnecessary print statements

jiljames committed:
Merge pull request #5 from jiljames/working

Removed unnecessary print statements

Commit:
Added a change to loss calculation

jiljames committed:
Merge pull request #6 from jiljames/working

Added a change to loss calculation

Commit:
Removed more unneeded prints

jiljames committed:
Merge pull request #7 from jiljames/working

Removed more unneeded prints

Commit:
Updated Readme and changed some parameters

jiljames committed:
Merge pull request #8 from jiljames/working

Updated Readme and changed some parameters

Commit:
Removed a blank line causing issues in valid file

jiljames committed:
Merge pull request #9 from jiljames/working

Removed a blank line causing issues in valid file

Commit:
Added some changes to do an obama speech run.

jiljames committed:
Merge pull request #10 from jiljames/working

Added some changes to do an obama speech run.

Commit:
Something is off about the training files...

Commit:
Had to chamnge sequence length in dataloader too

jiljames committed:
Merge pull request #11 from jiljames/working

Working

Commit:
Worked out a sizing issue with the train/valid files

Commit:
Worked out sizing issue with negative files

jiljames committed:
Merge pull request #12 from jiljames/working

Working

Commit:
change sequence length in neg file in dataloader

jiljames committed:
Merge pull request #13 from jiljames/working

change sequence length in neg file in dataloader

Commit:
More sizing problems fixed

jiljames committed:
Merge pull request #14 from jiljames/working

More sizing problems fixed

Commit:
First version of the changed seqgan main

jiljames committed:
Merge pull request #15 from jiljames/working

First version of the changed seqgan main

Commit:
fixed a naming problem

jiljames committed:
Merge pull request #16 from jiljames/working

fixed a naming problem

Commit:
Fixed file naming issue

jiljames committed:
Merge pull request #17 from jiljames/working

Fixed file naming issue

Commit:
Added random seeds and limited epochs for testing

jiljames committed:
Merge pull request #18 from jiljames/working

Added random seeds and limited epochs for testing

Commit:
Trying to fix testing conditions

jiljames committed:
Merge pull request #19 from jiljames/working

Trying to fix testing conditions

Commit:
Adding limited argparser functionality

jiljames committed:
Merge pull request #20 from jiljames/working

Adding limited argparser functionality

Commit:
Fixed parser issue

jiljames committed:
Merge pull request #21 from jiljames/working

Fixed parser issue

Commit:
Fixed parser issue

jiljames committed:
Merge pull request #22 from jiljames/working

Fixed parser issue

Commit:
Added user specified n to parser

jiljames committed:
Merge pull request #23 from jiljames/working

Added user specified n to parser

Commit:
Fixed an issue with sequence length in dataloader

jiljames committed:
Merge pull request #24 from jiljames/working

Fixed an issue with sequence length in dataloader

Commit:
Fixed vocab size change

jiljames committed:
Merge pull request #25 from jiljames/working

Fixed vocab size change

Commit:
Fixed sequence length

jiljames committed:
Merge pull request #26 from jiljames/working

Fixed sequence length

Commit:
Small bug fix

jiljames committed:
Merge pull request #27 from jiljames/working

Small bug fix

mhrmm commented on SeqGAN/obama/obama_helper.py:
```diff
@@ -0,0 +1,99 @@
+import json
+import random
+
+# Helper functions for converting between Haiku to DataLoader form
```
Haiku?

mhrmm replied:
There is a lot of code reuse between the obama_helper.py and haiku_helper.py. I would take both directories out of SeqGan and merge the two *_helper.py filles into a top_level dataprocessing.py file that reuses as many functions as possible. I'd also put the data (haiku.train.txt, haiku.valid.txt, obama.train.txt, obama.valid.txt) into the topmost data directory.

jiljames replied:
I fixed the data processing files issue by adding a dataloader_helper.py file and removing the old helper files. I thought though that keeping the data files (haiku.train.txt, haiku.valid.txt, obama.train.txt, obama.valid.txt) in the subdirectories makes sense because it makes each application a separate module. What do you think? Does that work with the way I restructured it? @mhrmm

mhrmm commented on SeqGAN/dataloader.py:
```diff
@@ -1,10 +1,12 @@
 import numpy as np
 
+np.random.seed(88)
```
You probably don't want these seed statements scattered across multiple files... it'll make it quite difficult to disable to random seed when you eventually want randomness. Just have one set at the top-level main.py (in this case, module_sequence_gan.py).

jiljames replied:
Got it!

mhrmm reviewed (commented):
Hi Jillian, I made some preliminary restructuring suggestions. To modify the pull request, all you need to do is change your local branch and then push the changes to origin. No need to make another pull request. Send me an @ message when it's ready to look at.

Commit:
Removed unneeded random seeds

Commit:
Added dataloader_helper.py and generated_example.py. Modified gitignore

Commit:
deleted old helper files that were replaced by dataloader_helper.py

Commit:
Added generated files to git ignore

jiljames committed:
Merge pull request #28 from jiljames/working

Cleaned up some issues mark pointed out. Created new dataloader files.

Commit:
Added more to .gitignore

jiljames committed:
Merge pull request #29 from jiljames/working

Added more to .gitignore

Commit:
Updated readme

jiljames committed:
Merge pull request #30 from jiljames/working

Updated readme

jiljames commented:
Just a heads up: I also changed the readme to match where I'm at now and added some of the files that are generated with dataloader_helper.py to the .gitignore since I thought adding them to the github made for more clutter. @mhrmm

mhrmm commented on SeqGAN/dataloader_helper.py:
```diff
@@ -0,0 +1,171 @@
+    
+
+
+def obama_to_ls(file):
```
Can this be unified with haiku_to_ls by adding a couple parameters?

mhrmm commented on SeqGAN/dataloader_helper.py:
```diff
@@ -0,0 +1,171 @@
+        int_to_word[integer_form] = token
+        integer_form += 1
+
+    return word_to_int, int_to_word, integer_form
```
integer_form is not really necessary to return, since you can get the same thing from len(word_to_int) or len(int_to_word).

mhrmm commented on SeqGAN/dataloader_helper.py:
```diff
@@ -0,0 +1,171 @@
+        train_ls = haiku_to_ls("haiku/haiku.train.txt")
+        valid_ls = haiku_to_ls("haiku/haiku.valid.txt")
+    else:
+        Print("Application must be haiku or obama")
```
Needs to be lowercase "print"

mhrmm reviewed (commented):
Great! The next important thing to do for this PR is to create unit tests for some of the new code you've provided, particularly the stuff in dataloader_helper.py (which you could rename datautil.py to make it a bit more concise). Here's an example unit test file (test_dataloader_helper.py):

```
import unittest
from dataloader_helper import creat_dicts

class TestDatautil(unittest.TestCase):

    def test_creat_dicts(self):
        w2i, i2w, num_toks = creat_dicts(['i am a haiku.\nOr am I not a haiku?\nCount and find out.'])
        assert w2i == {'\n': 1, ' ': 2, '.': 3, '?': 4, 'C': 5, 'I': 6, 
                       'O': 7, 'a': 8, 'd': 9, 'f': 10, 'h': 11, 'i': 12, 
                       'k': 13, 'm': 14, 'n': 15, 'o': 16, 'r': 17, 
                       't': 18, 'u': 19}
        assert i2w == {1: '\n', 2: ' ', 3: '.', 4: '?', 5: 'C', 6: 'I', 
                       7: 'O', 8: 'a', 9: 'd', 10: 'f', 11: 'h', 12: 'i', 
                       13: 'k', 14: 'm', 15: 'n', 16: 'o', 17: 'r', 
                       18: 't', 19: 'u'}
        assert num_toks == 20

if __name__ == "__main__":
	unittest.main()

```

For the functions that take files as arguments, you might want to restructure them so that they take the input streams as their arguments (to make them easier to unit test).

mhrmm commented:
Oh, and to get the above unit test to work, the creat_dicts function needs to sort the tokens before indexing them (this is likely a good idea anyway, but it makes testing simple as well).

Commit:
For running haiku on char level

jiljames committed:
Merge pull request #31 from jiljames/working

For running haiku on char level

Commit:
Small changes for char level haiku running

jiljames committed:
Merge pull request #32 from jiljames/working

Small changes for char level haiku running

Commit:
Changes some of datautil.py and added a test

jiljames committed:
Merge pull request #33 from jiljames/working

Changed some of datautil.py and added a test

Commit:
Changed some sizing issues in module file

Commit:
Added initial changes to testing and rollout files

Commit:
Backup for rollout.py

Commit:
rollout change

jiljames committed:
Merge pull request #34 from jiljames/working

Added changes to testing and rollout files

Commit:
fixed bug in rollout.py

jiljames committed:
Merge pull request #35 from jiljames/working

fixed bug in rollout.py

Commit:
naming issue fixed

jiljames committed:
Merge pull request #36 from jiljames/working

naming issue fixed

Commit:
testing a change to rollouts init

jiljames committed:
Merge pull request #37 from jiljames/working

testing a change to rollout's init

Commit:
fixed typo

Commit:
Added simple Bleuscore calculation in generated_example.py

jiljames committed:
Merge pull request #38 from jiljames/working

Added bleuscore and fixed typo

Commit:
Input file is now split between train, valid, test

Commit:
Removed comments. Removed cut off for under length 10.

Commit:
Deleted the raw .valid and .train files since they arent needed

jiljames committed:
Merge pull request #39 from jiljames/working

Input file is now split between train, valid, test

Commit:
Removed unneeded file

jiljames committed:
Merge pull request #40 from jiljames/working

Removed unneeded file

jiljames commented:
@mhrmm Okay! If you'd like to take a look I made the changes we talked about. I should add though, the corpus_bleu function takes a very long time ~20 mins on my machine. (I haven't run it on your machine because I think I will need you to do a install the nltk python library before doing so.)

mhrmm reviewed (commented):
LGTM. Go ahead and merge.
